test/intervals/test_intervals.jl

```diff
@@ -0,0 +1,239 @@
+module TestIntervals
+
+using FactCheck
+using Distributions
+using Bio
+using Bio.Intervals
+
+
+# Generate an array of n random Interval{Int} object. With sequence names
+# samples from seqnames, and intervals drawn to lie in [1, maxpos].
+function random_intervals(seqnames, maxpos::Int, n::Int)
+    seq_dist = Categorical(length(seqnames))
+    strand_dist = Categorical(2)
+    length_dist = Normal(1000, 1000)
+
+    intervals = Array(Interval{Int}, n)
+    for i in 1:n
+        intlen = ceil(Int, rand(length_dist))
+        first = rand(1:maxpos-intlen)
+        last = first + intlen - 1
+        strand = rand(strand_dist) == 1 ? STRAND_POS : STRAND_NEG
+        intervals[i] = Interval{Int}(seqnames[rand(seq_dist)],
+                                     first, last, strand, i)
+    end
+    return intervals
+end
+
+
+# A simple interval intersection implementation to test against.
+function simple_intersection(intervals_a, intervals_b)
+    sort!(intervals_a)
+    sort!(intervals_b)
+
+    intersections = Any[]
```

Comment:
Instead of `Any[]`, we could use `(eltype(intervals_a), elemtype(intervals_b))[]` to make the type of this Array tighter. But that's just a silly optimization.